oracle/price-feeder/oracle/oracle.go

```diff
@@ -417,6 +420,46 @@ func (o *Oracle) GetParams(ctx context.Context) (oracletypes.Params, error) {
 	return queryResponse.Params, nil
 }
 
+func (o *Oracle) GetJailedStateCache(ctx context.Context, currentBlockHeight int64) (bool, error) {
+	if !o.jailCache.IsOutdated(currentBlockHeight) {
+		return o.jailCache.isJailed, nil
+	}
+
+	isJailed, err := o.GetJailedState(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	o.jailCache.Update(currentBlockHeight, isJailed)
+	return isJailed, nil
+}
+
+// GetJailedState returns the current on-chain jailing state of the validator
+func (o *Oracle) GetJailedState(ctx context.Context) (bool, error) {
+	grpcConn, err := grpc.Dial(
+		o.oracleClient.GRPCEndpoint,
+		// the Cosmos SDK doesn't support any transport security mechanism
+		grpc.WithInsecure(),
+		grpc.WithContextDialer(dialerFunc),
+	)
+	if err != nil {
+		return false, fmt.Errorf("failed to dial Cosmos gRPC service: %w", err)
+	}
+
+	defer grpcConn.Close()
```

Comment:
Can we keep the grpcConn open?

Reply:
I followed the behavior here:
https://github.com/sei-protocol/sei-chain/blob/7f5681547fd498c5fbdd9fc8af45e200caae7e33/oracle/price-feeder/oracle/oracle.go#L408
where it closes the connection after the GetParams, I guess we could open one at the start of oracle and leave it open and then only close it when we terminate the price feeder?